AI hologram speech is now visible, outside runechat

DrDuckedGoose opened the pull request:
## About The Pull Request

AI hologram speech is now visible, swapping the ``send_speech()`` proc to ``say()``.

## Why It's Good For The Game

Makes interacting with AI holograms easier. Currently, people often forget to check runechat, to see what AI holograms are saying.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![image](https://user-images.githubusercontent.com/40559528/184520644-3f775fe6-0fd5-4578-bc20-0cefcbac214c.png)

![image](https://user-images.githubusercontent.com/40559528/184520639-d0e7000f-d572-4821-99ad-521d64200d93.png)

![image](https://user-images.githubusercontent.com/40559528/184520672-ac4db2de-2a35-4208-9833-65e6829dd31e.png)

![image](https://user-images.githubusercontent.com/40559528/184520674-a63f49f3-1155-4198-9724-147281dcae27.png)

</details>

## Changelog
:cl:
tweak: Swap ``holopad_talk`` to use ``say`` instead of ``send_speech``
add: Holopad machines now have a reference to AI-Holograms, for manipulation. 
/:cl:

DrDuckedGoose committed:
initial - 14 8 22

- adds ``holo`` as a referable variable to hologram.dm
- Changes AI-Hologram speech to use ``say``, to avoid runechat confusion.

itsmeow reviewed (changes requested):
You should block relayed speech from the hologram itself

PowerfulBacon commented on code/game/machinery/hologram.dm:
```diff
@@ -509,7 +511,7 @@ For the other part of the code, check silicon say.dm. Particularly robot talk.*/
 /obj/machinery/holopad/proc/move_hologram(mob/living/user, turf/new_turf)
 	if(LAZYLEN(masters) && masters[user])
-		var/obj/effect/overlay/holo_pad_hologram/holo = masters[user]
+		holo = masters[user]
```
Storing it up there will cause hard-dels won't it, since the reference will continue after this proc completes. I don't see anywhere this is ever unset

DrDuckedGoose committed:
review changes - 18 8 22

- Handle Bacon's point about hard deleting variables.

Archanial commented on code/modules/mob/living/silicon/ai/say.dm:
```diff
@@ -41,7 +41,7 @@
 			padloc = "(UNKNOWN)"
 		src.log_talk(message, LOG_SAY, tag="HOLOPAD in [padloc]")
-		send_speech(message, 7, T, MODE_ROBOT, message_language = language)
+		T?.holo?.say("[message]", MODE_ROBOT, language = language)
```
T cannot be null

DrDuckedGoose committed:
It literally doesn't matter, thanks though? - 21 8 22

- Remove safe access operator from T, it never should have been there, to be fair

Archanial commented on code/modules/mob/living/silicon/ai/say.dm:
```diff
@@ -41,7 +41,7 @@
 		else
 			padloc = "(UNKNOWN)"
 		src.log_talk(message, LOG_SAY, tag="HOLOPAD in [padloc]")
```
Doesn't this get logged twice? (say should log also)

DrDuckedGoose committed:
Fix hearing self & logging - 31 8 22

that_was_easy.wav

DrDuckedGoose committed:
Update say.dm

- remove redundant logging shit

DrDuckedGoose committed:
Update say.dm

itsmeow commented on code/modules/mob/living/silicon/ai/ai.dm:
```diff
@@ -951,6 +951,9 @@
 				jobpart = "[I.assignment]"
 
+	//AIs can't hear holograms
+	if(istype(speaker, /obj/effect/overlay/holo_pad_hologram))
```
you should check if the hologram specifically belongs to the pad the AI is currently using, so they can hear other holograms still.

itsmeow reviewed (changes requested):
in
`/obj/machinery/holopad/Hear`
change
`if(speaker && LAZYLEN(masters) && !radio_freq)`
to
`if(speaker && LAZYLEN(masters) && !radio_freq && speaker != holo)`

DrDuckedGoose committed:
Fix AI Inclusion - 19 9 22

- Thx bby

DrDuckedGoose committed:
Update hologram.dm
